src/install-opentofu.ps1

```diff
@@ -0,0 +1,502 @@
+    [int]  $ExitCode
+    [bool] $PrintUsage
+    ExitCodeException([string] $message, [int] $exitCode) : base($message) {
+        $this.ExitCode = $exitCode
+        $this.PrintUsage = $false
+    }
+    ExitCodeException([string] $message, [int] $exitCode, [bool] $printUsage) : base($message) {
+        $this.ExitCode = $exitCode
+        $this.PrintUsage = $printUsage
+    }
+}
+
+class InvalidArgumentException : ExitCodeException {
+    InvalidArgumentException([string] $message) : base($message, $exitCodeInvalidArgument, $true) {
+
+    }
+}
+
+class InstallMethodNotSupportedException : ExitCodeException {
+    InstallMethodNotSupportedException([string] $message) : base($message, $exitCodeInstallMethodNotSupported, $false) {
+
+    }
+}
+
+class InstallFailedException : ExitCodeException {
+    InstallFailedException([string] $message) : base($message, $exitCodeInstallFailed, $false) {
+    }
+}
+
+function logInfo() {
+    param(
+        $message
+    )
+    Write-Information "${blue}${message}${normal}"
+}
+
+function logWarning() {
+    param(
+        $message
+    )
+    Write-Warning "${orange}${message}${normal}"
+}
+
+function logError() {
+    param(
+        $message
+    )
+    try
+    {
+        [Console]::Error.WriteLine("${red}${message}${normal}")
+    } catch {}
+}
+
+function tempdir() {
+    $tempPath = [System.IO.Path]::GetTempPath()
+    $randomName = [System.IO.Path]::GetRandomFileName()
+    $path = Join-Path $tempPath $randomName
+    New-Item -Path $path -ItemType directory
+}
+
+function unpackStandalone() {
+    logInfo "Unpacking ZIP file to $installPath..."
+    try
+    {
+        New-Item -Path $installPath -ItemType directory -Force
+    }
+    catch
+    {
+        $msg = $_.ToString()
+        throw [InstallFailedException]::new("Failed to create target directory at ${installPath}. (${msg})")
+    }
+    $prevProgressPreference = $global:ProgressPreference
+    $global:ProgressPreference = 'SilentlyContinue'
+    try
+    {
+        logInfo "Unzipping $internalZipFile to $installPath..."
+        Expand-Archive -LiteralPath $internalZipFile -DestinationPath $installPath -Force
+    }
+    catch
+    {
+        $msg = $_.ToString()
+        throw [InstallFailedException]::new("Failed to unzip to ${installPath}. (${msg})")
+    }
+    finally
+    {
+        $global:ProgressPreference = $prevProgressPreference
+    }
+
+    if ($skipChangePath) {
+        return
+    }
+    try {
+        if ($allUsers) {
+            logInfo "Updating system PATH variable..."
+            $target = [EnvironmentVariableTarget]::Machine
+        } else {
+            logInfo "Updating user PATH variable..."
+            $target = [EnvironmentVariableTarget]::User
+        }
+        $currentPath = [Environment]::GetEnvironmentVariable("Path", $target)
+        if (!($currentPath.Contains($installPath))) {
+            [Environment]::SetEnvironmentVariable("Path", $currentPath + ";${installPath}", $target)
```

Comment:
question: are you sure this permanently changes the env var and not just temporarily for the current session? 🤔

Reply:
Hi @peterdeme , I have tested this method and it does indeed add the path permanently to the user's or the system profile depending on the `$target` on Windows 11. The SO question is where I found the hint for this API call, I just adapted the call to our needs.

![Screenshot of the Windows path editor window](https://github.com/opentofu/get.opentofu.org/assets/86970079/acf09985-3941-405b-a5d9-d959ba2c8c98)